Extraction/Classes/ARSwitchView.m

```diff
@@ -32,228 +32,239 @@ @implementation ARSwitchView
+    
+    NSMutableArray *selectionIndicatorConstraints = [[NSMutableArray alloc] initWithCapacity:titles.count];
+    NSMutableArray *buttons = [[NSMutableArray alloc] initWithCapacity:titles.count];
+    
+    for (NSString *title in titles) {
+        UIButton *button = [self createButtonWithTitle:title];
+        [self.buttonContainer addSubview:button];
+        [buttons addObject:button];
+        
+        // These constraints will be activated and deactivated to move the indicator.
+        NSLayoutConstraint *indicatorConstraint = [self.selectionIndicator alignLeadingEdgeWithView:button predicate:@"0"];
+        [selectionIndicatorConstraints addObject:indicatorConstraint];
+    };
     
-    // These constraints will be activated and deactivated to move the indicator.
-    NSLayoutConstraint *indicatorConstraint = [self.selectionIndicator alignLeadingEdgeWithView:button predicate:@"0"];
-    [selectionIndicatorConstraints addObject:indicatorConstraint];
-  };
-  
-  self.buttons = buttons;
-  self.selectionIndicatorConstraints = selectionIndicatorConstraints;
-  [NSLayoutConstraint deactivateConstraints:self.selectionIndicatorConstraints];
-  
-  [self.buttons.firstObject alignLeadingEdgeWithView:self.buttonContainer predicate:@"0"];
-  [self.buttons.lastObject alignTrailingEdgeWithView:self.buttonContainer predicate:@"0"];
-  
-  [UIView spaceOutViewsHorizontally:self.buttons predicate:@"0"];
-  [UIView alignTopAndBottomEdgesOfViews:[(NSArray<UIView *> *)self.buttons arrayByAddingObject:self.buttonContainer]];
-  [UIView equalWidthForViews:[(NSArray<UIView *> *)self.buttons arrayByAddingObject:self.selectionIndicator]];
-  
-  // Try to maintain the selection. With React Native this allows us to change a
-  // title at runtime without resetting the selection.
-  if (self.selectedIndex < titles.count) {
-    self.selectedIndex = self.selectedIndex;
-  } else {
-    self.selectedIndex = 0;
-  }
+    self.buttons = buttons;
+    self.selectionIndicatorConstraints = selectionIndicatorConstraints;
+    [NSLayoutConstraint deactivateConstraints:self.selectionIndicatorConstraints];
+    
+    [self.buttons.firstObject alignLeadingEdgeWithView:self.buttonContainer predicate:@"0"];
+    [self.buttons.lastObject alignTrailingEdgeWithView:self.buttonContainer predicate:@"0"];
+    
+    [UIView spaceOutViewsHorizontally:self.buttons predicate:@"0"];
+    [UIView alignTopAndBottomEdgesOfViews:[(NSArray<UIView *> *)self.buttons arrayByAddingObject:self.buttonContainer]];
+    [UIView equalWidthForViews:[(NSArray<UIView *> *)self.buttons arrayByAddingObject:self.selectionIndicator]];
+    
+    // Try to maintain the selection. With React Native this allows us to change a
+    // title at runtime without resetting the selection.
+    if (self.selectedIndex < titles.count) {
+        self.selectedIndex = self.selectedIndex;
+    } else {
+        self.selectedIndex = 0;
+    }
+    
+    [self _setSelectedIndex:self.selectedIndex animated:NO];
```

Comment:
sure, but in the case of 0, shouldn't that be captured in the global variable? (`self.selectedIndex`). Otherwise we're moving it to 0, but it's not properly saved.